test/NuGet.Clients.Tests/NuGet.VisualStudio.Implementation.Test/Extensibility/VsPackageSourceProviderTests.cs

```diff
@@ -57,12 +63,14 @@ public void GetSources_WhenUnknownExceptionIsThrown_ThrowsKnownException()
             packageSourceProvider.Setup(x => x.LoadPackageSources())
                 .Throws(originalException);
 
-            var vsPackageSourceProvider = new VsPackageSourceProvider(sourceRepositoryProvider.Object);
+            var vsPackageSourceProvider = new VsPackageSourceProvider(sourceRepositoryProvider.Object, telemetryProvider.Object);
 
             var actualException = Assert.Throws<InvalidOperationException>(() => vsPackageSourceProvider.GetSources(includeUnOfficial: true, includeDisabled: true));
 
             Assert.Equal(originalException.Message, actualException.Message);
             Assert.Same(originalException, actualException.InnerException);
+
+            telemetryProvider.Verify(p => p.PostFault(originalException, typeof(VsPackageSourceProvider).FullName, nameof(VsPackageSourceProvider.GetSources), It.IsAny<IDictionary<string, object>>()));
```

Comment:
I understand your intention we don't add new unit test for this new telemetry exception handler.
But just sake of exception handling can we still add just 1-2 unit test where you intentionally throw some random exception and make sure something is captured in telemetry context? My point is if it doesn't catch intentional/known exception then how it capture unknown one?

Reply:
I added a whole bunch of tests now, for everything that is testable.